### PR TITLE
Reference of tools.build and tools.files

### PR DESCRIPTION
**Pull request by lasote**

### CODE REVIEW

**Review by czoido — APPROVED** [review]

File: reference/tools/files/basic.rst [hunk 1]
```diff
@@ -77,6 +191,61 @@ conan.tools.files.unzip()
...
+.. warning::
+
+    This tool collects the libraries searching directly inside the package folder and returns them in no specific order. If libraries are
+    inter-dependent, then ``package_info()`` method should order them to achieve correct linking order.
+
+Usage:
+
+.. code-block:: python
+
+    from conans import tools
```
Comment (memsharded) [inline reference/tools/files/basic.rst:247]: Wrong import and usage

File: reference/tools/files/basic.rst [hunk 2]
```diff
@@ -77,6 +191,61 @@ conan.tools.files.unzip()
...
+
+    def package_info(self):
+        self.cpp_info.libdirs = ["lib", "other_libdir"]  # Deafult value is 'lib'
+        self.cpp_info.libs = tools.collect_libs(self)
```
Comment (memsharded) [inline reference/tools/files/basic.rst:251]: Wrong usage

File: reference/tools/files/downloads.rst [hunk 3]
```diff
@@ -17,10 +17,57 @@ conan.tools.files.ftp_download()
 .. autofunction:: ftp_download
 
 
+Usage:
+
+.. code-block:: python
+
+    from conans import tools
```
Comment (memsharded) [inline reference/tools/files/downloads.rst:24]: Wrong import

File: reference/tools/files/downloads.rst [hunk 4]
```diff
@@ -17,10 +17,57 @@ conan.tools.files.ftp_download()
...
+
+    def source(self):
+        tools.ftp_download('ftp.debian.org', "debian/README")
```
Comment (memsharded) [inline reference/tools/files/downloads.rst:27]: Wrong usage

File: reference/tools/files/packaging.rst [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        packager.patterns.src == []
+        packager.patterns.build == []
+        packager.patterns.res == []
+        packager.patterns.framework == []
+
+Usage:
+
+.. code:: python
+
+        from conans import ConanFile
```
Comment (memsharded) [inline reference/tools/files/packaging.rst:37]:
```suggestion
        from conan import ConanFile
```

**Review by memsharded — CHANGES_REQUESTED** [review]

**Review by memsharded — APPROVED** [review]

**Pull request merged**